csharp/Repl/RoslynScriptEngine.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Scripting;
+using Microsoft.CodeAnalysis.Scripting;
+using Microsoft.Spark.CSharp.Core;
+using Microsoft.Spark.CSharp.Sql;
+using Razorvine.Pickle;
+using Razorvine.Serpent;
+
+[assembly: InternalsVisibleTo("ReplTest")]
+namespace Microsoft.Spark.CSharp
+{
+    /// <summary>
+    /// Class that holds all objects shared in Roslyn runtime context.
+    /// </summary>
+    public class SparkCLRHost
+    {
+        public SparkContext sc;
+        public SqlContext sqlContext;
+    }
+
+    /// <summary>
+    /// A C# script engine based on Roslyn(https://github.com/dotnet/roslyn).
+    /// </summary>
+    public class RoslynScriptEngine : IScriptEngine
+    {
+        private int dumpSeq = 0;
+        private ScriptState<object> previousState;
+        private readonly SparkConf sparkConf;
+        private readonly SparkContext sc;
+        private readonly SparkCLRHost host;
+        private readonly ParseOptions options;
+
+        internal readonly string compilationDumpDirectory;
+
+        public RoslynScriptEngine(SparkContext sc)
+        {
+            this.sc = sc;
+            sparkConf = sc.GetConf();
```

Comment:
@skaarthik @kai-zeng 

Agreed that SqlContext should be constructed outside ScriptEngine.

Reasons for introducing GetConf() instead of not passing sparkConf from Main method:
* There is also a public `getConf()` method defined in `SparkContext` ([link](https://github.com/apache/spark/blob/branch-1.6/core/src/main/scala/org/apache/spark/SparkContext.scala#L256))
* `getConf()`  returns a **copy** of this SparkContext's configuration (the configuration **cannot** be changed at runtime).
* As SparkConf can be got from SparkContext via `GetConf()`, code will be clean if unnecessary params won't be passed in.

In addition, since SparkConf is kind of readonly (just a copy), passing SparkConf instead of `localDirPath` is much flexible as other Spark configuration might be used in the future.